src/main/scala/com/fulcrumgenomics/util/DelimitedDataParser.scala

```diff
@@ -0,0 +1,128 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.util
+
+import com.fulcrumgenomics.FgBioDef._
+import dagr.commons.reflect.ReflectionUtil
+import dagr.commons.util.LazyLogging
+
+import scala.reflect.runtime.{universe => ru}
+
+/**
+  * Represents a row of parsed data.  Provides methods for accessing values in a type-safe
+  * way either via apply() methods for non-optional fields or via get for optional fields.
+  */
+class Row (private val headerIndices: Map[String,Int], private val fields: Array[String], val trim: Boolean) {
+  /* Internal method to pull a value out of the field array by index and trim it if requested. */
+  private def value(index: Int) = {
+    if (index > fields.length-1) throw new IndexOutOfBoundsException(s"Invalid column index supplied: ${index}")
+    val string = fields(index)
+    if (trim) string.trim else string
+  }
+
+  /** Fetches a value of the desired type by 0-based column index. */
+  def apply[A](index: Int)(implicit tag: ru.TypeTag[A]): A = {
+    val c = ReflectionUtil.typeTagToClass(tag)
+    ReflectionUtil.constructFromString(c, c, value(index)).get.asInstanceOf[A]
+  }
+
+  /** Fetches a value of the desired type by column name. */
+  def apply[A](column: String)(implicit tag: ru.TypeTag[A]): A = apply(headerIndices.apply(column))
```

Comment:
Fixed.  I'd made a bunch of "exploratory" changes because the compiler was giving me an inscrutable error message at some point so I was being more explicit everywhere.